### PR TITLE
Onboard certificate changes

### PR DESCRIPTION
**Pull request by nithyal1**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 1]
```diff
@@ -6948,4 +6974,797 @@ else if (matchingPolicies.stream().anyMatch(p-> p.equals("r"+itemName)) || match
...
+		     * @return
+		     */
+		    private ResponseEntity<String> createCertMetadataAndPolicy(SSLCertificateRequest sslCertificateRequest, UserDetails userDetails, int containerId) {
+		    	
+		    	// Policy Creation
+				boolean isPoliciesCreated;
+				CertResponse enrollResponse = new CertResponse();
+				try {
+					String metadataJson = populateSSLCertificateMetadataForOnboard(sslCertificateRequest, userDetails,containerId);
+					isPoliciesCreated = createPolicies(sslCertificateRequest, tokenUtils.getSelfServiceToken());
```
Comment (Ukothur1) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:7374]: Iam seeing logs are added to createPolicies() method, but those are not showing in PR

File: tvaultapi/src/main/resources/application.properties [hunk 2]
```diff
@@ -567,4 +567,18 @@ OIDCAuthController.searchGroupsInAzure.notes=To search groups in AAD
...
+nclm.mock=false
+
+SSLCertificateController.onboardcert.value=Bulk onboard certificates from NCLM to Tvault
+SSLCertificateController.onboardcert.notes= Bulk onboard certificates from NCLM to Tvault
+SSLCertificateController.onboardsinglecert.value=onboard single certificate from NCLM to Tvault
+SSLCertificateController.onboardsinglecert.notes=onboard single certificate from NCLM to Tvault
+pacbot.endpoint.getToken=https://pacbot.t-mobile.com/api/auth/oauth/token?grant_type=client_credentials
+pacbot.endpoint.getallcertificates=https://pacbot.t-mobile.com/api/asset/v1/list/assets
+pacbot.client.id=c2dcefc6-b30e-4a88-83e3-5fab05ab5821
+pacbot.client.secret=8cXsBuXUX8gn8JG7wS5v3jTTSeSQsKFayudKy2wmKCjJb
```
[the remark this answers is not in the document]
Comment (nithyal1) [reply tvaultapi/src/main/resources/application.properties]: Removed

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 3]
```diff
@@ -6948,4 +6974,797 @@ else if (matchingPolicies.stream().anyMatch(p-> p.equals("r"+itemName)) || match
...
+		                 put(LogMessage.APIURL, ThreadLocalContext.getCurrentMap().get(LogMessage.APIURL)).
+		                 build()));
+		    	 
+		    	 String dnsStr = ObjectUtils.isEmpty(certObject.get("subjectaltname_dns"))?"":certObject.get("subjectaltname_dns").getAsString();
+		    	 if(!StringUtils.isEmpty(dnsStr)) {
+		    	 String[] dnslist = Arrays.stream(dnsStr.split(","))
+		    		        .map(String::trim)
+		    		        .toArray(String[]::new); 
+		    	 sslCertificateRequest.setDnsList(dnslist); 
+		    	 }
```
[the remark this answers is not in the document]
Comment (nithyal1) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java]: Yes.these dns names are overwritten from the nclm response and hence removed

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 4]
```diff
@@ -6948,4 +6974,797 @@ else if (matchingPolicies.stream().anyMatch(p-> p.equals("r"+itemName)) || match
...
+		                    build()));
+		        }}
+
+		        
+		        CertificateData certDetails = null;
+		        String nclmAccessToken = getNclmToken();	
+		                
+		        //Get Certificate Details
+				if(!StringUtils.isEmpty(nclmAccessToken)) {
+		        for (int i = 1; i <= retrycount; i++) {
```
[the remark this answers is not in the document]
Comment (nithyal1) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java]: done

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java (hunk 1 above)
Comment (nithyal1) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:7368]: the log changes are already merged in the last PR

**Review by Ukothur1 — APPROVED** [review]

**Pull request merged**